cli/npm/mod.rs

```diff
@@ -127,6 +132,21 @@ impl GlobalNpmPackageResolver {
         "Unstable use of npm specifiers. The --unstable flag must be provided."
       )
     }
+
+    if self.no_npm {
```

Comment:
```suggestion
    if self.no_npm && !packages.is_empty() {
```

Maybe just in case like the above.

Reply:
That's a code smell for me, we should assert that `packages` is not empty in that function, 3 out of 4 callers verify it already

Reply:
I added an assert